aviary/subsystems/propulsion/engine_scaling.py

```diff
@@ -130,107 +144,86 @@ def compute(self, inputs, outputs):
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.THRUST,
-            'thrust_net_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.THRUST_MAX,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.THRUST_MAX,
-            'thrust_net_max_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
-            'fuel_flow_rate_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.ELECTRIC_POWER_IN,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.ELECTRIC_POWER_IN,
-            'electric_power_in_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.NOX_RATE,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.NOX_RATE,
-            'nox_rate_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.SHAFT_POWER,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.SHAFT_POWER,
-            'shaft_power_unscaled',
-            rows=r, cols=r,
-            val=1.0)
-
-        self.declare_partials(
-            Dynamic.Mission.SHAFT_POWER_MAX,
-            Aircraft.Engine.SCALE_FACTOR,
-            rows=r, cols=c,
-            val=1.0)
-        self.declare_partials(
-            Dynamic.Mission.SHAFT_POWER_MAX,
-            'shaft_power_max_unscaled',
-            rows=r, cols=r,
-            val=1.0)
+        for variable in engine_variables:
+            if variable not in skip_variables:
+                if variable is FUEL_FLOW:
+                    self.declare_partials(
+                        Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+                        Aircraft.Engine.SCALE_FACTOR,
+                        rows=r,
+                        cols=c,
+                        val=1.0,
+                    )
+                    self.declare_partials(
+                        Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+                        'fuel_flow_rate_unscaled',
+                        rows=r,
+                        cols=r,
+                        val=1.0,
```

Comment:
should this be -1?

Reply:
or do we not need any of these `val=1`, since we will calculate it below?

Reply:
Looks like we didn't need to specify val for any of these, now removed